userspace/libscap/engine/gvisor/runsc.cpp

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright (C) 2022 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include <unistd.h>
+#include <sys/wait.h>
+
+#include "gvisor.h"
+
+namespace scap_gvisor {
+
+namespace runsc {
+
+std::string root_path;
+std::string trace_session_path;
+constexpr size_t max_line_size = 8192;
+
+void init(std::string root, std::string trace_session)
+{
+	root_path = root;
+	trace_session_path = trace_session;
+}
```

Comment:
Let's keep the state in the gvisor engine instead of global variables. The namespaced functions could then get those two parameters in the arguments.